language/move-lang/src/unit_test/plan_builder.rs

```diff
@@ -0,0 +1,297 @@
+                    (*aloc, "Assigned in this attribute"),
+                ]),
+            };
+            args
+        }
+        EA::Parameterized(nm, attributes) => {
+            assert!(
+                nm.value == known_attributes::TestingAttributes::TEST,
+                "ICE: We should only be parsing a raw test attribute"
+            );
+            args.extend(
+                attributes
+                    .iter()
+                    .flat_map(|attr| parse_test_attribute(context, attr)),
+            );
+            args
+        }
+    }
+}
+
+fn parse_failure_attribute(
+    context: &mut Context,
+    sp!(aloc, expected_attr): &E::Attribute,
+) -> Option<ExpectedFailure> {
+    use E::{AttributeValue_ as EAV, Attribute_ as EA, Value_ as EV};
+    match expected_attr {
+        EA::Name(nm) => {
+            assert!(
+                nm.value == known_attributes::TestingAttributes::EXPECTED_FAILURE,
+                "ICE: We should only be parsing a raw expected failure attribute"
+            );
+            Some(ExpectedFailure::Expected)
+        }
+        EA::Assigned(_, sp!(assign_loc, _)) => {
+            context.env.add_error(vec![
+                (
+                    *assign_loc,
+                    "Invalid expected failure code assignment".into(),
+                ),
+                (
+                    *aloc,
+                    format!(
+                        "Expect an #[expected_failure({}=...)] attribute for abort code assignment",
+                        known_attributes::TestingAttributes::CODE_ASSIGNMENT_NAME
+                    ),
+                ),
+            ]);
+            None
+        }
+        EA::Parameterized(sp!(_, nm), attrs) => {
+            if attrs.len() != 1 {
+                context.env.add_error(vec![
+                    (*aloc, format!("Invalid #[expected_failure(...)] attribute, expected 1 argument but found {}", attrs.len())),
+                ]);
+                return None;
+            }
+            assert!(
+                nm == known_attributes::TestingAttributes::EXPECTED_FAILURE,
+                "ICE: expected failure attribute must have the right name"
+            );
+            match attrs.last().unwrap() {
+                sp!(_, EA::Assigned(sp!(_, nm), value))
+                    if nm == known_attributes::TestingAttributes::CODE_ASSIGNMENT_NAME =>
+                {
+                    match value {
+                        sp!(_, EAV::Value(sp!(_, EV::U64(u)))) => {
+                            Some(ExpectedFailure::ExpectedWithCode(*u))
+                        }
+                        sp!(_, EAV::Value(sp!(_, EV::U8(u)))) => {
+                            Some(ExpectedFailure::ExpectedWithCode(*u as u64))
+                        }
+                        sp!(_, EAV::Value(sp!(_, EV::U128(u)))) if *u <= std::u64::MAX as u128 => {
+                            Some(ExpectedFailure::ExpectedWithCode(*u as u64))
+                        }
```

Comment:
bit too magic for me, these only come up from the `u8` and `u128` suffixes being explicit. I would probably jsut error and tell the user it has to be a valid u64 value

Reply:
One issue that I had was that for some reason all of the attribute values in e.g., `#[expected_failure(abort_code=5)]` are getting parsed as/converted to u128s somewhere up ahead in the pipeline. Let me take a look at how to fix that, and once I put that up I'll update this to only accept u64s

Reply:
Ah I think I dropped it when converting from Parser to Expansion attributes. We should undo that and just carry along the inferred num

Reply:
I've updated so it now requires a non-annotated value by tagging this in the `AttributeValue_::Value` variant in expansion. Let me know if it looks good to you.